addons/mail/static/src/web_editor_emoji_command/web_editor_emoji_command.js

```diff
@@ -0,0 +1,63 @@
+/** @odoo-module */
+
+import { _t } from 'web.core';
+import Wysiwyg from 'web_editor.wysiwyg';
+import { getRangePosition } from '@web_editor/js/editor/odoo-editor/src/utils/utils';
+
+Wysiwyg.include({
+    async willStart() {
+        await this._super();
+        this.messaging = undefined;
+        if (owl.Component.env.services.messaging) {
+            owl.Component.env.services.messaging.get().then(messaging => {
+                this.messaging = messaging;
+                this.messaging.messagingBus.addEventListener('add_emoji', this._onAddEmoji.bind(this));
+            });
+        }
+    },
+    destroy() {
+        this._super();
+        if (!this.messaging || !this.messaging.messagingBus) {
+            return;
+        }
+        this.messaging.messagingBus.removeEventListener('add_emoji', this._onAddEmoji.bind(this));
+    },
+    _onAddEmoji(ev) {
+        this.odooEditor.historyResetLatestComputedSelection();
+        this.odooEditor.execCommand('insert', ev.detail.emoji);
+    },
+    /**
+     * @override
+     */
+    _getPowerboxOptions: function () {
+        const options = this._super();
+        if (!this.messaging || !this.messaging.messagingBus) {
+            return options;
+        }
+        options.categories.push({
+            name: _t('Mail'),
+            priority: 300,
+        });
```

Comment:
- Would use category `Messaging` (none of the already defined categories are appropriate)
- priority should be relatively high in `knowledge`, something as close as the `Knowledge` and `Media`, which are at 10 and 50 respectively.